pyproject.toml

```diff
@@ -92,3 +93,25 @@ module = [
   "traceloggingdynamic",
 ]
 ignore_missing_imports = true
+
+[tool.poe.tasks.install]
+shell = "for dir in . ni_measurementlink_generator/ examples/*/; do pushd $dir; poetry install; popd; done"
+
+[tool.poe.tasks.ni-python-styleguide]
+args = [
+  { name = "command", positional = true, required = true }
+]
+shell = "for dir in . ni_measurementlink_generator/ examples/*/; do pushd $dir; ni-python-styleguide $command; popd; done"
+
+[tool.poe.tasks.mypy]
+sequence = [
+  { cmd = "mypy -p ni_measurementlink_service" },
+  { cmd = "mypy -p ni_measurementlink_generator", cwd = "ni_measurementlink_generator" },
+  { shell = "for dir in examples/*/; do pushd $dir; mypy .; popd; done" }
```

Comment:
This is actually running the copy of mypy that is installed in the root venv, not the copy in `examples/sample_measurement/.venv` or whatever. Someone (@mshafer-NI?) asked about this and I gave the wrong answer. 

For most of the examples, it still works out, but for `game_of_life` and `sample_measurement` it currently generates additional warnings that don't occur when you cd to the example directory and run `poetry run mypy .`.

This also affects ni-python-styleguide, but it doesn't matter because using the example's pyproject.toml is sufficient.

Reply:
Yes, that was me.

We need to tell poetry to ignore the root .venv and make new .venv's (maybe clearing the VIRTUALENV_PATH would do this?), then here, get it to use the individual envs...

I would suggest posting a discussion on the poethepoet repo, the maintainer has been pretty responsive, and if someone else has figured this out they may have ideas.

Reply:
> We need to tell poetry to ignore the root .venv and make new .venv's (maybe clearing the VIRTUALENV_PATH would do this?), then here, get it to use the individual envs...

Yes, unsetting `VIRTUAL_ENV` seems like a potential path forward. There are a couple of caveats with this approach, which I listed in the discussion post.

> I would suggest posting a discussion on the poethepoet repo

https://github.com/nat-n/poethepoet/discussions/166